### PR TITLE
Removed unnecessary/broken code within models. Seed refactored

### PR DESCRIPTION
**Pull request by fire-pls**
seed can still be refactored further, I just included comments in code to explain

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -6,11 +6,12 @@ class Event < ApplicationRecord
   validates :game, :title, :address, :description, presence: true
 
   after_create :initialize_event
-  before_update :ensure_gamesize
+  before_save :ensure_gamesize
```
Comment (fire-pls) [inline app/models/event.rb]: gamesize needs to be proper all times an event is saved to a database, not just when it's updated

File: app/models/event.rb [hunk 2]
```diff
@@ -6,11 +6,12 @@ class Event < ApplicationRecord
...
 
   def add_user(usr)
-    if self.game.max_players < self.players.length && !attending?(usr)
-      self.players << Player.create(user:usr,event:self)
+    if self.players.length < self.game.max_players && !attending?(usr)
```
Comment (fire-pls) [inline app/models/event.rb]: old code is more dry, but semantically broken
before
add player to event playerlist IF game max length is LESS THAN current playercount

after
add player to event IF current playercount is LESS THAN game max length

Comment (fire-pls) [reply app/models/event.rb]: use:
```ruby
if self.players.length < self.game.max_players && !attending?(usr)
  self.players << Player.create(user:usr,event:self)
  self.save
  return true
end
```

File: app/models/host.rb [hunk 3]
```diff
@@ -1,8 +1,4 @@
 class Host < ApplicationRecord
   belongs_to :user
   has_many :events
-
```
Comment (fire-pls) [inline app/models/host.rb]: Users table doesn't store usernames at the moment. this should be commented so we can use later

File: app/models/user.rb [hunk 4]
```diff
@@ -10,30 +10,17 @@ class User < ApplicationRecord
...
   has_many :players
-  after_create :initialize_host, :initialize_user
+  after_create :initialize_host
   validates :username, presence: true, :uniqueness => {
     :case_sensitive => false
   }
 
-
-  def self.get_by_username(usr)
-    #currently case sensitive
```
Comment (fire-pls) [inline app/models/user.rb]: Users table doesn't store usernames atm, just comment this for use later

File: app/models/user.rb [hunk 5]
```diff
@@ -10,30 +10,17 @@ class User < ApplicationRecord
...
-  end
-
   private
 
   def initialize_host
     Host.create(user_id:self.id)
   end
 
-  def initialize_user
-    self.level = 1.0
```
Comment (fire-pls) [inline app/models/user.rb]: the table sets these as the defaults already, KEEP this but call with
```ruby
before_create :initialize_user
```
to ensure every new user cannot be injected with custom settings

File: app/models/host.rb (hunk 3 above)
Comment (jdgc) [reply app/models/host.rb]: fixed

File: app/models/user.rb (hunk 4 above)
Comment (jdgc) [reply app/models/user.rb]: fixed on master

**Pull request merged**